.golangci.yml

```diff
@@ -94,72 +96,26 @@ linters-settings:
     # if it's called for subdir of a project it can't find external interfaces. All text editor integrations
     # with golangci-lint call it on a directory with the changed file.
     check-exported: false
-  nakedret:
-    # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
-    max-func-lines: 30
-  prealloc:
-    # Report preallocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
-    # True by default.
-    simple: true
-    range-loops: true # Report preallocation suggestions on range loops, true by default
-    for-loops: true # Report preallocation suggestions on for loops, false by default
   gocritic:
     disabled-checks:
       - ifElseChain
       - singleCaseSwitch
       - appendAssign
-linters:
-  enable:
-    - deadcode
-    - errcheck
-    - gocritic
-    - goimports
-    - golint
-    - govet
-    - gosimple
-    - interfacer
-    - lll
-    - maligned
-    - misspell
-    - staticcheck
-    - structcheck
-    - ineffassign
-    - megacheck
-    - typecheck
-    - unconvert
-    - unparam
-    - unused
-    - varcheck
-  enable-all: false
-  disable:
-    - gochecknoglobals
-    - gochecknoinits
-    - goconst
-    - gosec
-    - nakedret
-    - prealloc
-    - scopelint
-  disable-all: false
-  presets:
-    - bugs
-    - unused
-  fast: false
 
 issues:
   # List of regexps of issue texts to exclude, empty list by default.
   # But independently from this option we use default exclude patterns,
   # it can be disabled by `exclude-use-default: false`. To list all
   # excluded by default patterns execute `golangci-lint run --help`
-  exclude: []
+  exclude:
+    - composite literal uses unkeyed fields
```

Comment:
Why remove this? This is very important one since we had issues with people reordering struct keys...

Reply:
govet was not being run, this PR makes it run. As a result, that particular warning showed up several dozens (hundreds?) of times and it wasn't immediately obvious whether we should do such a change, since it does hurt readability in many cases.

So let's go with this change as-is, and we can try removing this exclusion as a separate PR and see what it looks like.

Reply:
Hmmm OK. We will need to fix those violations, I think. It is a real problem since we are guilty of removing/renumbering fields without taking into account unkeyed structs.